tests/contrib/urllib3/test_urllib3.py

```diff
@@ -0,0 +1,385 @@
+import mock
+import pytest
+import urllib3
+
+from ddtrace import config
+from ddtrace.constants import ANALYTICS_SAMPLE_RATE_KEY
+from ddtrace.contrib.urllib3 import patch
+from ddtrace.contrib.urllib3 import unpatch
+from ddtrace.ext import errors
+from ddtrace.ext import http
+from ddtrace.pin import Pin
+from tests import TracerTestCase
+from tests.opentracer.utils import init_tracer
+
+
+# socket name comes from https://english.stackexchange.com/a/44048
+SOCKET = "httpbin.org"
+URL_200 = "http://{}/status/200".format(SOCKET)
+URL_500 = "http://{}/status/500".format(SOCKET)
+
+
+class BaseUrllib3TestCase(TracerTestCase):
+    """Provides the setup and teardown for patching/unpatching the urllib3 integration"""
+
+    def setUp(self):
+        super(BaseUrllib3TestCase, self).setUp()
+
+        patch()
+        self.http = urllib3.PoolManager()
+        Pin.override(urllib3.connectionpool.HTTPConnectionPool, tracer=self.tracer)
+
+    def tearDown(self):
+        super(BaseUrllib3TestCase, self).tearDown()
+        unpatch()
+
+
+class TestUrllib3(BaseUrllib3TestCase):
+    def test_HTTPConnectionPool_traced(self):
+        """Tests that requests made from the HTTPConnectionPool are traced"""
+        pool = urllib3.connectionpool.HTTPConnectionPool(SOCKET)
+        # Test a relative URL
+        r = pool.request("GET", "/status/200")
+        assert r.status == 200
+        spans = self.tracer.writer.pop()
```

Comment:
```suggestion
        spans = self.pop_spans()
```

(also for all the other usages)